library/src/main/java/com/novoda/downloadmanager/lib/DownloadService.java

```diff
@@ -154,6 +157,13 @@ public void onCreate() {
         mExecutor = factory.createExecutor();
     }
 
+    private DownloadClientReadyChecker getDownloadClientReadyChecker() {
+        if (!(getApplication() instanceof DownloadClientReadyChecker)) {
```

Comment:
But it is very difficult to pass dependencies into the `DownloadService` :disappointed: